src/sympc/protocol/falcon/falcon.py

```diff
@@ -63,3 +65,77 @@ def __eq__(self, other: Any) -> bool:
             return False
 
         return True
+
+    @staticmethod
+    def mul(x, y, session) -> ReplicatedSharedTensor:
+        """Master method for multiplication.
+
+        Performs Falcon's mult implementation, gets and reshares mul results and distributes shares.
+        This needs to be improved in future, it relies on orchestrator being a trusted third party.
+        Falcon, requires parties to be able to communication between each other.
+
+        Args:
+            x (MPCTensor): Secret
+            y (MPCTensor): Another secret
+            session (Session): Session the tensors belong to
+
+        Returns:
+            shares (ReplicatedSharedTensor): Shares in terms of ReplicatedSharedTensor.
+
+        Raises:
+            ValueError: Raised when number of parties are not three.
+            NotImplementedError: Raised when implementation not present
+
+        """
+        if not len(session.parties) == 3:
+            raise ValueError("Falcon requires 3 parties")
+
+        result = None
+
+        if session.protocol.security_type == "semi-honest":
+            result = Falcon.mul_implementation(x, y)
+
+        elif session.protocol.security_type == "malicious":
+            raise NotImplementedError(
+                "Multiplication has not been implemented for malicious setting"
+            )
+
+        else:
+            raise NotImplementedError(
+                "Multiplication has not been implemented for "
+                + session.protocol.security_type
+                + " setting"
+            )
+
+        vals = [result[0].get(), result[1].get(), result[2].get()]
+        # Add random mask to vals
+        shares = MPCTensor.generate_shares(
+            sum(vals), 3, config=Config(encoder_base=1, encoder_precision=0)
+        )
+        shares = ReplicatedSharedTensor.distribute_shares(shares, x.session)
+        return shares
+
+    @staticmethod
+    def mul_implementation(x, y) -> List[ReplicatedSharedTensor]:
```

Comment:
can we have a different name for this function this sounds little bit off

Reply:
multiplication_protocol?